### PR TITLE
workspace: add naming convention lints

### PR DESCRIPTION
**Pull request by hinto-janai**
### What
Adds:

- https://rust-lang.github.io/rust-clippy/master/index.html#/upper_case_acronyms
- https://doc.rust-lang.org/rustc/lints/listing/warn-by-default.html#non-camel-case-types

to enforce the [Rust naming convention](https://rust-lang.github.io/api-guidelines/naming.html#casing-conforms-to-rfc-430-c-case).

---

Changes `RandomXVM` -> `RandomXVm` and some other things. To be fair `RandomX` is an uncommon case but here are some other VM examples:

- https://reth.rs/docs/reth/revm/struct.Evm.html
- https://reth.rs/docs/reth/core/rpc/types/trace/parity/struct.VmTrace.html

### CODE REVIEW

File: clippy.toml [hunk 1]
```diff
@@ -0,0 +1 @@
+upper-case-acronyms-aggressive = true
```
Comment (hinto-janai) [inline clippy.toml:1]: This is a clippy option for `upper_case_acronyms`:

> Triggers if there is more than one uppercase char next to each other

So it catches things like `CamelCASE`.

Comment (Boog900) [reply clippy.toml:1]: How does `RandomXVm` pass this lint, the `XV` are uppercase characters next to each other.

**Review by Boog900 — APPROVED** [review]
Approved just the one question.

File: clippy.toml (hunk 1 above)
Comment (hinto-janai) [reply clippy.toml:1]: [There's special conditions for acronyms](https://github.com/rust-lang/rust-clippy/blob/master/clippy_lints/src/upper_case_acronyms.rs#L59-L72), either:

- 3 uppercase chars back-to-back
- 2 ending uppercase chars

In `RandomXVm`'s case, the last `m` resets the uppercase char count to 0...

**Pull request merged**